### PR TITLE
rewrite

### PR DESCRIPTION
**Pull request by 1000ch**
- [x] rewrite with es6
- [x] add the option to fix string on save
- [x] remove the missing option (how to apply message type -> default level?)

related to https://github.com/AtomLinter/linter-plus/issues/58

### CODE REVIEW

**Comment by iam4x** [comment]
@1000ch I'll take time to review this on Monday, thank's for the hard work :+1:

File: index.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  let text   = editor.getText();
+
+  if (isMissingConfig && onlyConfig()) {
+    return;
+  }
+
+  editor.setText(jscs.fixString(text, path).output);
+};
+
+export const provideLinter = () => {
```
Comment (iam4x) [inline index.js]: I love ES6 syntax, but the function context binding is not needed here.

`export function provideLinter() {`

^ makes more sense I think.

**Comment by steelbrain** [comment]
Just a sidenote, but I think it'll make this PR a **lot** more readable.
Babel allows property initializers, both of these will work
```js
class Something{
  static staticVar = 1
  privateVar = 1
}
```
so that way this linter class could be written as
```js
class LinterJSCS{
  static Config = {
    executablePath: {type: "string", default: "", description: "Executable Path"}
  }
  static activate(){
    console.log("Package activated")
  }
  static provideLinter(){
    return {
    // .. linter stuff here
    }
  }
}
module.exports LinterJSCS
```

**Comment by iam4x** [comment]
:ship:

**Comment by 1000ch** [comment]
@iam4x Thank you :)

**Comment by williamboman** [comment]
:raised_hands: Thanks!

**Pull request merged**